src/WebJobs.Extensions.DurableTask/DurableOrchestrationContext.cs

```diff
@@ -422,6 +422,18 @@ public override void ContinueAsNew(object input, bool preserveUnprocessedEvents)
         {
             this.ThrowIfInvalidAccess();
 
+            // These check can be removed once the storage provider supports extended timers.
+            // https://github.com/Azure/azure-functions-durable-extension/issues/14
+            if (retryOptions.FirstRetryInterval > TimeSpan.FromDays(MaxTimerDurationInDays))
+            {
+                throw new ArgumentException($"Retry intervals must not exceed {MaxTimerDurationInDays} days.", nameof(retryOptions.FirstRetryInterval));
+            }
+
+            if (retryOptions.MaxRetryInterval > TimeSpan.FromDays(MaxTimerDurationInDays))
+            {
+                throw new ArgumentException($"Retry intervals must not exceed {MaxTimerDurationInDays} days.", nameof(retryOptions.MaxRetryInterval));

```

Comment:
As you mentioned, this is a breaking change, which we can't allow for Durable 1.x. I think the best path forward is to  just set (overwrite) large retry intervals with our 6 day maximum. In the PR for Durable 2.x, we can throw here and/or throw in the property setters.